media/live-seekable-range.js

```diff
@@ -1,68 +1,88 @@
-const video = document.createElement('video');
+const video = document.querySelector('video');
 
 const mediaSource = new MediaSource();
 video.src = URL.createObjectURL(mediaSource);
   
 mediaSource.addEventListener('sourceopen', function() {
   URL.revokeObjectURL(video.src);
 
-  const videoUrl = 'https://storage.googleapis.com/media-session/sample.webm';
-  const sourceBuffer = mediaSource.addSourceBuffer('video/webm; codecs="vp9"');
-
-  // Makes video behave like a live stream.
-  mediaSource.duration = +Infinity;
+  mediaSource.addSourceBuffer('video/webm; codecs="vp9"');
 
   log('Fetching video init segment...');
-  fetch(videoUrl, { headers: { range: 'bytes=0-299' } })
-  .then(response => response.arrayBuffer())
-  .then(data => {
-
-    // Append video init segment.
-    sourceBuffer.appendBuffer(data);
-    sourceBuffer.addEventListener('updateend', function() {
-      logTimeRanges();
-
-      log('Fetching video segment that starts at 3 seconds...');
-      fetch(videoUrl, { headers: { range: 'bytes=567140-1196488' } })
-      .then(response => response.arrayBuffer())
-      .then(data => {
+  fetchAndAppendMediaSegment('bytes=0-299');
```

Comment:
nit: to not confuse folks with "Media" segment for what is technically the bytestream segment corresponding to an MSE "Initialization Segment", perhaps rename this method to "fetchAndAppendSegment"?